tests/datadriven/test_reads.py

```diff
@@ -24,7 +59,58 @@ def getDataModelClass(self):
     def getProtocolClass(self):
         return protocol.GAReadGroupSet
 
-    def testFixMe(self):
+    def testGetReadAlignmentsBothRefs(self):
+        # test that querying by both referenceName and referenceId fails
+        with self.assertRaises(exceptions.BadReadsSearchRequestBothRefs):
+            readGroupSet = self._gaObject
+            for readGroup in readGroupSet._readGroups:
+                list(readGroup.getReadAlignments("a", 5, 0, None))
+
+    def testGetReadAlignments(self):
+        # test that searching with no arguments succeeds
+        readGroupSet = self._gaObject
+        for i, readGroup in enumerate(readGroupSet._readGroups):
```

Comment:
I'm slightly uneasy about relying on the ordering of the readGroups here. Perhaps we could just keep a dictionary mapping readGroupId to the list of reads, rather than two linked arrays (i.e. _readGroups and _reads)? That is, we have a dictionary self._reads which maps readGroupId to the reads in the SamFile.